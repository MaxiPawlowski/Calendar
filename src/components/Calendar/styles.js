import { createStyles } from '@material-ui/core';

export default () =>
  createStyles({
    root: {
      display: 'flex',
      flexWrap: 'wrap',
      padding: '1%',
      flexBasis: '80%'
    },
    days: {
      flex: `0 0 calc(${100 / 7}% - 10px)`,
      margin: '5px',
      padding: '5px',
      backgroundColor: '#e4f0f5',
      border: '1px solid rgba(50,50,50,.2)',
      borderRadius: '4px',
      textAlign: 'center',
    },
  });
