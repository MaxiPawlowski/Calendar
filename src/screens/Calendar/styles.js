import { createStyles } from '@material-ui/core';

export default () => 
  createStyles({
    root: {
      display: 'flex',
    },
    sidebar:{
      flexBasis: "20%",
      padding: "0px 1%",
      borderRight: '1px solid rgba(0, 0, 0, 0.12)',
      display: 'flex',
      flexDirection: 'column',
      justifyContent: 'space-between',
    }
  });
