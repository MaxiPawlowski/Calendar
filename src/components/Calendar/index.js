import React from "react";
import CalendarDay from 'components/CalendarDay'
import { withStyles } from '@material-ui/styles';
import {
  addDays,
  compareDates,
  daysInMonth,
  daysName,
  lastDateOfCalendar,
  firstDateOfCalendar
} from "utils/date"
import PropTypes from 'prop-types';
import styles from "./styles";

const ReminderForm = ({ classes, reminders, calendarDate, handleDayExpand }) => {
  const days = [];
  const firstItemDate = firstDateOfCalendar(calendarDate);
  const lastItemDate = lastDateOfCalendar(calendarDate);

  for (let i = 1; i <= firstItemDate.getDay(); i++) {
    days.unshift(
      <CalendarDay
        key={days.length}
        reminders={reminders.filter(({ date }) => compareDates(date, addDays(firstItemDate, -i)))}
        date={addDays(firstItemDate, -i)}
      />
    );   
  }
  
  for (let i = 1; i <= daysInMonth( calendarDate ); i++) {
    const calendarItemDate = new Date(`${calendarDate.month}/${i}/${calendarDate.year}`);
    days.push(
      <CalendarDay
        key={days.length}
        handleDayExpand={handleDayExpand}
        reminders={reminders.filter(({ date }) => compareDates(date, calendarItemDate))}
        date={calendarItemDate} 
      />
    );    
  }
    
  for (let i = 1; i <= 6 - lastItemDate.getDay(); i++) {
    days.push(
      <CalendarDay
        key={days.length}
        reminders={reminders.filter(({ date }) => compareDates(date, addDays(lastItemDate, i)))}
        date={addDays(lastItemDate, i)}
      />
    );   
  }

  return (
    <div className={classes.root}>
      {daysName.map((day, index) => (
        <div key={index} className={classes.days}>
          {day}
        </div>
      ))}
      {days}
    </div>
  );
};

ReminderForm.propTypes = {
  classes: PropTypes.shape({
    root: PropTypes.string.isRequired,
    days: PropTypes.string.isRequired,
  }).isRequired,

  calendarDate: PropTypes.shape({
    year: PropTypes.number.isRequired,
    month: PropTypes.number.isRequired,
  }).isRequired,

  reminders: PropTypes.arrayOf(PropTypes.shape({
    title: PropTypes.string.isRequired,
    color: PropTypes.string.isRequired,
    date: PropTypes.instanceOf(Date).isRequired,
    city: PropTypes.string.isRequired
  })).isRequired,

  handleDayExpand: PropTypes.func.isRequired,
};

export default withStyles(styles)(ReminderForm);
