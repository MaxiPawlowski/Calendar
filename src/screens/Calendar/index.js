import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { withStyles } from '@material-ui/core/styles';
import { createReminder, changeCalendarDate } from 'actions';
import ReminderForm from 'components/ReminderForm';
import CalendarComponent from 'components/Calendar';
import RemindersDialog from 'components/RemindersDialog';
import { MuiPickersUtilsProvider, DatePicker } from '@material-ui/pickers';
import { dateToCalendar, calendarToDate } from "utils/date"
import DateFnsUtils from '@date-io/date-fns';
import styles from './styles'

class Calendar extends Component {
  constructor(props){
    super(props);
    this.state = {
      open: false,
      dialog:{
        date: new Date(Date.now()),
        reminders: []
      },
    }
  }

  handleSave = (values) => {
    const { addReminder } = this.props;
    addReminder(values);
  }

  handleDialogOpen = dialog => {
    this.setState({
      open:true,
      dialog,
    });
  };

  handleDialogClose = () => {
    this.setState({
      open:false,
      dialog: {
        date: new Date(Date.now()),
        reminders: []
      }
    });
  };

  render() {
    const { classes, reminders, calendarDate, changeDate } = this.props;
    const { open, dialog } = this.state;

    return (
      <div className={classes.root}>
        <div className={classes.sidebar}>
          <ReminderForm handleSave={this.handleSave} />
          <MuiPickersUtilsProvider utils={DateFnsUtils}>
            <DatePicker
              openTo="year"
              views={["year", "month"]}
              label="Calendar Date"
              value={calendarToDate(calendarDate)}
              onChange={(date) => changeDate(dateToCalendar(date))}
            />
          </MuiPickersUtilsProvider>
        </div>
        <CalendarComponent calendarDate={calendarDate} reminders={reminders} handleDayExpand={this.handleDialogOpen} />
        <RemindersDialog data={dialog} open={open} handleClose={this.handleDialogClose} />
      </div>
    );
  }
}

const mapStateToProps = (state) => ({
  reminders: state.remindersReducer.reminders,
  calendarDate: state.dateReducer,
});

const mapDispatchToProps = (dispatch) => ({
  addReminder: (reminder) => dispatch(createReminder(reminder)),
  changeDate: (date) => dispatch(changeCalendarDate(date))
});


Calendar.propTypes = {
  classes: PropTypes.shape({
    root: PropTypes.string.isRequired,
  }).isRequired,

  reminders: PropTypes.arrayOf(PropTypes.shape({
    title: PropTypes.string.isRequired,
    color: PropTypes.string.isRequired,
    date: PropTypes.instanceOf(Date).isRequired,
    city: PropTypes.string.isRequired
  })).isRequired,

  calendarDate: PropTypes.shape({
    year: PropTypes.number.isRequired,
    month: PropTypes.number.isRequired,
  }).isRequired,

  addReminder: PropTypes.func.isRequired,
  changeDate: PropTypes.func.isRequired
};

export const StyledApp = withStyles(styles)(Calendar);

export default connect(mapStateToProps, mapDispatchToProps)(StyledApp);
