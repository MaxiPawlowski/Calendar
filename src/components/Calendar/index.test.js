import React from 'react';
import { shallow } from 'enzyme';
import Calendar from "./index";

describe('Calendar', () => {
  let wrapper;

  describe('when rendered', () => {
    beforeAll(() => {
      wrapper = shallow(
        <Calendar
          reminders={[]}
          handleDayExpand={()=>{}}
          deleteReminder={()=>{}}
          calendarDate={{
            year: 2019,
            month: 10
          }}
        />
      );
    });

    it('should render correctly', () => {
      expect(wrapper.dive()).toMatchSnapshot();
    });
  });
});
