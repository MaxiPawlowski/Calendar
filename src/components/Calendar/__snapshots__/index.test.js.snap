// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Calendar when rendered should render correctly 1`] = `
<div
  className="ReminderForm-root-1"
>
  <div
    className="ReminderForm-days-2"
    key="0"
  >
    Sunday
  </div>
  <div
    className="ReminderForm-days-2"
    key="1"
  >
    Monday
  </div>
  <div
    className="ReminderForm-days-2"
    key="2"
  >
    Tuesday
  </div>
  <div
    className="ReminderForm-days-2"
    key="3"
  >
    Wednesday
  </div>
  <div
    className="ReminderForm-days-2"
    key="4"
  >
    Thursday
  </div>
  <div
    className="ReminderForm-days-2"
    key="5"
  >
    Friday
  </div>
  <div
    className="ReminderForm-days-2"
    key="6"
  >
    Saturday
  </div>
  <WithStyles(ReminderForm)
    date={2019-09-29T03:00:00.000Z}
    key="1"
    reminders={Array []}
  />
  <WithStyles(ReminderForm)
    date={2019-09-30T03:00:00.000Z}
    key="0"
    reminders={Array []}
  />
  <WithStyles(ReminderForm)
    date={2019-10-01T03:00:00.000Z}
    handleDayExpand={[Function]}
    key="2"
    reminders={Array []}
  />
  <WithStyles(ReminderForm)
    date={2019-10-02T03:00:00.000Z}
    handleDayExpand={[Function]}
    key="3"
    reminders={Array []}
  />
  <WithStyles(ReminderForm)
    date={2019-10-03T03:00:00.000Z}
    handleDayExpand={[Function]}
    key="4"
    reminders={Array []}
  />
  <WithStyles(ReminderForm)
    date={2019-10-04T03:00:00.000Z}
    handleDayExpand={[Function]}
    key="5"
    reminders={Array []}
  />
  <WithStyles(ReminderForm)
    date={2019-10-05T03:00:00.000Z}
    handleDayExpand={[Function]}
    key="6"
    reminders={Array []}
  />
  <WithStyles(ReminderForm)
    date={2019-10-06T03:00:00.000Z}
    handleDayExpand={[Function]}
    key="7"
    reminders={Array []}
  />
  <WithStyles(ReminderForm)
    date={2019-10-07T03:00:00.000Z}
    handleDayExpand={[Function]}
    key="8"
    reminders={Array []}
  />
  <WithStyles(ReminderForm)
    date={2019-10-08T03:00:00.000Z}
    handleDayExpand={[Function]}
    key="9"
    reminders={Array []}
  />
  <WithStyles(ReminderForm)
    date={2019-10-09T03:00:00.000Z}
    handleDayExpand={[Function]}
    key="10"
    reminders={Array []}
  />
  <WithStyles(ReminderForm)
    date={2019-10-10T03:00:00.000Z}
    handleDayExpand={[Function]}
    key="11"
    reminders={Array []}
  />
  <WithStyles(ReminderForm)
    date={2019-10-11T03:00:00.000Z}
    handleDayExpand={[Function]}
    key="12"
    reminders={Array []}
  />
  <WithStyles(ReminderForm)
    date={2019-10-12T03:00:00.000Z}
    handleDayExpand={[Function]}
    key="13"
    reminders={Array []}
  />
  <WithStyles(ReminderForm)
    date={2019-10-13T03:00:00.000Z}
    handleDayExpand={[Function]}
    key="14"
    reminders={Array []}
  />
  <WithStyles(ReminderForm)
    date={2019-10-14T03:00:00.000Z}
    handleDayExpand={[Function]}
    key="15"
    reminders={Array []}
  />
  <WithStyles(ReminderForm)
    date={2019-10-15T03:00:00.000Z}
    handleDayExpand={[Function]}
    key="16"
    reminders={Array []}
  />
  <WithStyles(ReminderForm)
    date={2019-10-16T03:00:00.000Z}
    handleDayExpand={[Function]}
    key="17"
    reminders={Array []}
  />
  <WithStyles(ReminderForm)
    date={2019-10-17T03:00:00.000Z}
    handleDayExpand={[Function]}
    key="18"
    reminders={Array []}
  />
  <WithStyles(ReminderForm)
    date={2019-10-18T03:00:00.000Z}
    handleDayExpand={[Function]}
    key="19"
    reminders={Array []}
  />
  <WithStyles(ReminderForm)
    date={2019-10-19T03:00:00.000Z}
    handleDayExpand={[Function]}
    key="20"
    reminders={Array []}
  />
  <WithStyles(ReminderForm)
    date={2019-10-20T03:00:00.000Z}
    handleDayExpand={[Function]}
    key="21"
    reminders={Array []}
  />
  <WithStyles(ReminderForm)
    date={2019-10-21T03:00:00.000Z}
    handleDayExpand={[Function]}
    key="22"
    reminders={Array []}
  />
  <WithStyles(ReminderForm)
    date={2019-10-22T03:00:00.000Z}
    handleDayExpand={[Function]}
    key="23"
    reminders={Array []}
  />
  <WithStyles(ReminderForm)
    date={2019-10-23T03:00:00.000Z}
    handleDayExpand={[Function]}
    key="24"
    reminders={Array []}
  />
  <WithStyles(ReminderForm)
    date={2019-10-24T03:00:00.000Z}
    handleDayExpand={[Function]}
    key="25"
    reminders={Array []}
  />
  <WithStyles(ReminderForm)
    date={2019-10-25T03:00:00.000Z}
    handleDayExpand={[Function]}
    key="26"
    reminders={Array []}
  />
  <WithStyles(ReminderForm)
    date={2019-10-26T03:00:00.000Z}
    handleDayExpand={[Function]}
    key="27"
    reminders={Array []}
  />
  <WithStyles(ReminderForm)
    date={2019-10-27T03:00:00.000Z}
    handleDayExpand={[Function]}
    key="28"
    reminders={Array []}
  />
  <WithStyles(ReminderForm)
    date={2019-10-28T03:00:00.000Z}
    handleDayExpand={[Function]}
    key="29"
    reminders={Array []}
  />
  <WithStyles(ReminderForm)
    date={2019-10-29T03:00:00.000Z}
    handleDayExpand={[Function]}
    key="30"
    reminders={Array []}
  />
  <WithStyles(ReminderForm)
    date={2019-10-30T03:00:00.000Z}
    handleDayExpand={[Function]}
    key="31"
    reminders={Array []}
  />
  <WithStyles(ReminderForm)
    date={2019-10-31T03:00:00.000Z}
    handleDayExpand={[Function]}
    key="32"
    reminders={Array []}
  />
  <WithStyles(ReminderForm)
    date={2019-11-01T03:00:00.000Z}
    key="33"
    reminders={Array []}
  />
  <WithStyles(ReminderForm)
    date={2019-11-02T03:00:00.000Z}
    key="34"
    reminders={Array []}
  />
</div>
`;
